tests/integration/integration-test.sh

```diff
@@ -357,12 +357,32 @@ update_helm_for_workloads() {
 	rm update
 }
 
+update_for_self_hosted_notary() {
+	envsubst <tests/integration/update-self-hosted-notary.yaml >update
+	yq '. *+ load("ghcr-values")' -i update
+	yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' helm/values.yaml update
+	rm update
+	curl "https://notary.server:4443/v2/docker.io/securesystemsengineering/testimage/_trust/tuf/root.json" > root.json
+    SELF_HOSTED_NOTARY_PUBLIC_ROOT_KEY_ID=$(cat root.json | jq -r .signatures[0].keyid)
+    cat root.json | jq -r --arg KEYID $SELF_HOSTED_NOTARY_PUBLIC_ROOT_KEY_ID '.signed.keys | .[$KEYID] | .keyval.public' | base64 -d | openssl x509 -pubkey -noout > self_hosted_notary_root_key.pub
+    yq -i '(.application.validators.[] | select(.name == "self-hosted-notary") | .trustRoots.[] | select(.name=="default") | .key) |= load_str("self_hosted_notary_root_key.pub")' helm/values.yaml
+    rm self_hosted_notary_root_key.pub
+}
+
 enable_alerting() {
 	envsubst <tests/integration/update-alerting.yaml >update
 	yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' helm/values.yaml update
 	rm update
 }
 
+render_notary_host() {
```

Comment:
That's the "we can't yq the template, but need to add values there and sed is fragile, so we do this" part? If so, an explaining comment might be nice